### PR TITLE
SDAAP-113 Create a feeding service and parser for Bang Showbiz

### PR DESCRIPTION
**Pull request by marwoodandrew**

### CODE REVIEW

File: server/aap/io/feed_parsers/bang_parser.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        if len(located) > 0:
+            item["dateline"]["located"] = located[0]
+            item["dateline"]["text"] = format_dateline_to_locmmmddsrc(
+                located[0], item["versioncreated"], self.provider.get("source")
+            )
+        return True
+
+    def parse_news_identifier(self, item, tree):
+        parsed_el = self.parse_elements(tree.find("Identification/NewsIdentifier"))
+        item["uri"] = item["guid"] = "{}:{}".format(
```
Comment (MarkLark86) [inline server/aap/io/feed_parsers/bang_parser.py]: Is `NewsItemId` unique for each version?
Also, wondering if using a URN format might be a good standard to follow. In this case something like:
```python
"urn:newsml:{}:{}:{}".format(
    self.provider.get("current_url"),
    datetime.today().isoformat(),
    parsed_el["NewsItemId"]
)
```

Or even just the domain portion of self.provider.get("current_url")?

File: server/aap/io/feeding_services/bang.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        if not self.session:
+            self.session = requests.Session()
+
+        parser = self.get_feed_parser(provider)
+
+        items = []
+        for src in self.fields:
+            current_url = provider.get("config").get(src.get("id"))
+            if current_url:
+                provider["current_url"] = current_url
```
Comment (MarkLark86) [inline server/aap/io/feeding_services/bang.py]: Or even storing `src["id"]` on the provider, and using that in the URN generation?

**Review by MarkLark86 — APPROVED** [review]
Looks fine, just the one consideration/suggestion with the `guid`

File: server/aap/io/feeding_services/bang.py (hunk 2 above)
Comment (marwoodandrew) [reply server/aap/io/feeding_services/bang.py]: OK, URN now looks like 

    urn:newsml:music_url:2024-04-23T10:30:00+00:00:BS2BS3062256
    
dumping in the URL looked a bit yuk! From the id I can get which url it was.

File: server/aap/io/feed_parsers/bang_parser.py (hunk 1 above)
Comment (marwoodandrew) [reply server/aap/io/feed_parsers/bang_parser.py]: Sort of done!

**Review by MarkLark86 — APPROVED** [review]

**Pull request merged**